### PR TITLE
Splitter name

### PR DESCRIPTION
**Pull request by Christian-B**
Fixes https://github.com/SpiNNakerManchester/PACMAN/issues/459

Removes spltter name. and several string constants which where just the str(type(self))

Instead AbstractSplitterCommon now does str and repr based on the splitter type

Must be done at the same time as:
https://github.com/SpiNNakerManchester/sPyNNaker/pull/1216

tested by:
https://github.com/SpiNNakerManchester/IntegrationTests/pull/132

### CODE REVIEW

File: pacman/model/partitioner_splitters/splitter_one_to_one_legacy.py [hunk 1]
```diff
@@ -45,23 +45,17 @@ class SplitterOneToOneLegacy(AbstractSplitterCommon):
...
-    def __repr__(self):
-        return self.__str__()
-
     @overrides(AbstractSplitterCommon.set_governed_app_vertex)
     def set_governed_app_vertex(self, app_vertex):
         super().set_governed_app_vertex(app_vertex)
         self._vertex_slice = Slice(0, self._governed_app_vertex.n_atoms - 1)
         self._sdram = (
-            self._governed_app_vertex.get_sdram_used_by_atoms(
+             self._governed_app_vertex.get_sdram_used_by_atoms(
```
Comment (andrewgait) [inline pacman/model/partitioner_splitters/splitter_one_to_one_legacy.py:58]: github is saying an extra space has been added here which seems to mess up the formatting...

File: pacman/model/partitioner_splitters/abstract_splitters/abstract_splitter_common.py [hunk 2]
```diff
@@ -39,21 +35,18 @@ class AbstractSplitterCommon(object, metaclass=AbstractBase):
...
 
     def __str__(self):
-        return (
-            f"{self._splitter_name} governing app vertex"
-            f" {self._governed_app_vertex}")
+        try:
+            return (
+                f"{type(self).__name__} on {self._governed_app_vertex.label}")
+        except AttributeError:
+            return f"{type(self).__name__} no vertex"
```
Comment (andrewgait) [inline pacman/model/partitioner_splitters/abstract_splitters/abstract_splitter_common.py:49]: I would probably be more explicit here - "has no governed_app_vertex" or something like that.

**Review by andrewgait — CHANGES_REQUESTED** [review]
Looks fine, just a couple of minor points.

File: pacman/model/partitioner_splitters/splitter_one_to_one_legacy.py (hunk 1 above)
Comment (Christian-B) [reply pacman/model/partitioner_splitters/splitter_one_to_one_legacy.py:58]: reversed

**Review by andrewgait — APPROVED** [review]

**Pull request merged**